mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -34,8 +36,25 @@ import org.apache.spark.util.collection.OpenHashMap
 /**
  * Base trait for [[StringIndexer]] and [[StringIndexerModel]].
  */
-private[feature] trait StringIndexerBase extends Params with HasInputCol with HasOutputCol
-    with HasHandleInvalid {
+private[feature] trait StringIndexerBase extends Params with HasInputCol with HasOutputCol {
+  val SKIP_UNSEEN_LABEL: String = "skip"
+  val ERROR_UNSEEN_LABEL: String = "error"
+  val KEEP_UNSEEN_LABEL: String = "keep"
+  val supportedHandleInvalids: Array[String] =
+    Array(SKIP_UNSEEN_LABEL, ERROR_UNSEEN_LABEL, KEEP_UNSEEN_LABEL)
+
+  /**
+   * Param for how to handle unseen labels. Options are 'skip' (filter out rows with
+   * unseen labels), 'error' (throw an error), or 'keep' (map unseen labels with
```

Comment:
rephrase: "map unseen labels with indices [numLabels]" --> "put unseen labels in a special additional bucket, at index numLabels"